src/worker/hwdetect.rs

```diff
@@ -1,24 +1,44 @@
+use crate::common::parser::{format_parse_error, p_uint, NomResult};
+use anyhow::anyhow;
+
 use tako::common::resources::{CpuId, NumOfCpus, ResourceDescriptor};
 
-pub fn parse_range(line: &str) -> anyhow::Result<Vec<CpuId>> {
-    let line = line.trim();
-    let mut result = Vec::new();
-    for part in line.split(',') {
-        let part = part.trim();
-        if part.contains('-') {
-            let ids: Vec<&str> = part.split('-').collect();
-            if ids.len() != 2 {
-                anyhow::bail!("Invalid format of range");
-            }
-            for i in ids[0].parse()?..=ids[1].parse()? {
-                result.push(i);
-            }
-        } else {
-            let id: CpuId = part.parse()?;
-            result.push(id);
-        }
-    }
-    Ok(result)
+use nom::bytes::complete::tag;
+use nom::combinator::all_consuming;
+use nom::combinator::{map_res, opt};
+use nom::multi::separated_list1;
+use nom::sequence::{preceded, tuple};
+
+fn remove_whitespace(s: &str) -> String {
```

Comment:
Do not use this kind of method as it blindly removes spaces and makes the combinators uncomposable.
Use space1 from nom. Something like: terminated(xxx, space1) in places where space is allowed

Reply:
Updated at my local repo [ https://github.com/User3574/hyperqueue/tree/hwdetect_parser ].